### ISSUE TITLE
Library throws a runtime error on Node 12: SyntaxError: Unexpected token '?'

### ISSUE THREAD

**Issue by rachid-frontdev** [nfriedly/express-rate-limit#278]
### Description

hi ,
when i install dependency the command line show me this 

    const totalHits = (this.hits[key] ?? 0) + 1;
                                       ^
SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (C:\Users\Rachid\Desktop\javascript api\rateLimit.js:1
:20)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
for this directory ...\..\node_modules\express-rate-limit\dist\index.cjs:49

### Library version

6.0.3

### Node version

v12.22.6

### Module system

CommonJS

**Comment by nfriedly** [comment]
Hi @rachid-frontdev 

Version 6.x of express-rate-limit requires node.js 14 or newer. You can use 5.x with node.js 12 or, preferably, update to a more recent version of node.js.

It should warn you about this during installation - do you know if that happened? (Also, what version of npm are you using? `npm -v`)

**Comment by rachid-frontdev** [comment]
i have version npm 6.14.15

**Comment by nfriedly** [comment]
Ok, I realized that 6.0.3 had the min node.js change in the package.json, but not the package-lock.json. That may have been why it didn't stop you from installing the incompatible version. v6.0.4 now has the change in both.

**Comment by nfriedly** [comment]
Would you mind trying to install 6.0.4 and letting me know what happens?

**Comment by zhangxinyong12** [comment]
import * as rateLimit from 'express-rate-limit'

**Comment by diego-alvarez-by** [comment]
@nfriedly I'm facing same error, but I'm using node 14.18.2, npm 6.14.15 and typescript 4.2.2.

**Comment by nfriedly** [comment]
Hum, that's curious. It ought to be supported as of node 14.5 - https://node.green/#ES2020-features--nullish-coalescing-operator-----

But... it looks like ncc uses webpack to bundle things, and maybe webpack is parsing the code looking for `require()` calls...? I'm not sure. You might want to ask the Zeit folks if they have any idea what's going on here.

**Comment by nfriedly** [comment]
It looks like webpack uses acorn to parse js by default, and acorn does support `??` if you configure it with `{ecmaVersion: 2020}` or greater. But I'm not sure how to plumb that configuration all the way through.

**Comment by gamemaker1** [comment]
Maybe we should just change the line to 

```
const totalHits = (typeof this.hits[key] !== 'number' ? 0 : this.hits[key]) + 1
```

This way we'll be able to support node 12.9+, and the problem @diego-alvarez-by is facing should also be fixed.

**Comment by nfriedly** [comment]
Yeah, you're probably right @gamemaker1 - I thought about that, and then got lost in the weeds looking into how to enforce it with the linter, and then ran out of time. 

But really, we could just run the test suite on node 12.

**Comment by nfriedly** [comment]
There's something to be said for dragging the whole community forward, but probably better to just do the simple pragmatic thing of fixing of dragging this Library backwards a bit.

**Comment by gamemaker1** [comment]
We just released v6.3.0 with this fix, please update with `npm install express-rate-limit@latest` and let us know if this fixes the issue!
